src/hardware/driver/internal_adc.rs

```diff
@@ -79,4 +80,25 @@ impl InternalAdc {
         const OFFSET: f32 = 0.0; // Current sense offset         ToDo
         (ratio + OFFSET) * SCALE
     }
+
+    pub fn read_all(&mut self) -> [(f32, DriverMonitorChannel); 4] {
```

Comment:
Ah. This is not a convenient API. The return array looses all info about the values. And you have to reconstruct it manually from the indices. Better use `read_output_current/voltage()` directly.